brave/brave5/build.gradle

```diff
@@ -13,13 +13,17 @@ dependencies {
 }
 
 def brave6ProjectDir = "${rootProject.projectDir}/brave/brave6"
-tasks.compileJava.source "${brave6ProjectDir}/src/main/java"
-tasks.processResources.from "${brave6ProjectDir}/src/main/resources"
-tasks.compileTestJava.source "${brave6ProjectDir}/src/test/java"
-tasks.processTestResources.from "${brave6ProjectDir}/src/test/resources"
-tasks.sourcesJar.from "${brave6ProjectDir}/src/main/java"
-tasks.sourcesJar.from "${brave6ProjectDir}/src/main/resources"
-tasks.javadoc.source "${brave6ProjectDir}/src/main/java"
+
+tasks.register('generateSources', Copy.class) {
```

Comment:
curious do we prefer to use the convention like
```
def generateSources =  tasks.register('generateSources', Copy.class) {
...
}

tasks.named("compileJava").dependsOn(generateSources)
```

Reply:
Could you let me know if you are suggesting this convention? Or are you just asking? 😆 
If you are suggesting it, could you share the reason, please? 😉

Reply:
Maybe both, it's good to avoid some configuration time. And saw our native-image-config using similar way.
Notice this recently https://docs.gradle.org/current/userguide/task_configuration_avoidance.html

Reply:
Thank you for sharing the link. I wasn't aware that we were using an anti-pattern. 😓 Since we've employed this pattern across all our build files, let me create another PR to address this issue.
```
Avoid referencing a task by name. In the majority of cases, referencing a task by name is a fragile pattern and should be avoided. Although the task name is available on the TaskProvider, effort should be made to use references from a strongly typed model instead.
```